md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/TableColumnLineageAPI.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.wipro.ats.bdre.md.rest;
+
+
+import com.wipro.ats.bdre.ldg.GetDotForTable;
+import com.wipro.ats.bdre.md.api.base.MetadataAPIBase;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import java.security.Principal;
+
+
+/**
+ * Created by arijit on 1/9/15.
+ */
+@Controller
+@RequestMapping("/tabcollineage")
+public class TableColumnLineageAPI extends MetadataAPIBase {
+    private static final Logger LOGGER = Logger.getLogger(TableColumnLineageAPI.class);
+
+    /**
+     * This method returns the relationship between the processes linked to a batchId. It also generates the
+     * dot string for visualisation.
+     *
+     * @param batchId
+     * @return restWrapper It contains an instance of LineageInfo.
+     */
+    /*@RequestMapping(value = "/bybatch/{batchId}", method = RequestMethod.GET)
+    public
+    @ResponseBody
+    RestWrapper getByBatch(@PathVariable("batchId") String batchId, Principal principal) {
+        RestWrapper restWrapper = null;
+        try {
+            GetLineageByBatch getLineageByBatch = new GetLineageByBatch();
+            //insert args as received from the jsp
+            String[] args = {"-col", batchId};
+            GetDotForTable getDotForTable = new GetDotForTable();
+
+            String dot = getDotForTable.dotGeneratorWithCol(args);;
+
+            LineageInfo lineageInfo = new LineageInfo();
+            lineageInfo.setDot(dot);
+            lineageInfo.setBatchId(batchId);
+            restWrapper = new RestWrapper(lineageInfo, RestWrapper.OK);
+            LOGGER.info("Record with ID:" + batchId + " selected(getbyBatch) from Lineage by User:" + principal.getName());
+        } catch (Exception e) {
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }*/
+
+    /**
+     * This method is used to see the relationship between processes linked to a particular instanceExecId.
+     * It also generates the dot string for visualisation of the links.
+     *
+     * @param tableName
+     * @return restWrapper It contains an instance of LineageInfo.
+     */
+    @RequestMapping(value = {"", "/"}, method = RequestMethod.GET)
+    public
+    @ResponseBody
+    RestWrapper getByTableCol(@RequestParam(value = "tableName", defaultValue = "null") String tableName, @RequestParam(value = "colName", defaultValue = "null") String colName, Principal principal) {
+        RestWrapper restWrapper = null;
+        try {
+
+            if(colName != null && !(colName.trim().equals(""))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1132)